os_migrate/roles/prelude_common/tasks/main.yml

```diff
@@ -15,25 +15,24 @@
 # under the License.
 
 - name: get the current openstacksdk version
-  ansible.builtin.shell: |
-    set -eo pipefail
-    python3 -m pip show openstacksdk | grep Version | cut -d' ' -f 2
-  args:
-    executable: /bin/bash
+  community.general.pip_package_info:
   register: _installed_openstacksdk
-  changed_when: "_installed_openstacksdk.rc == 0"
+
+- name: set openstacksdk version
+  set_fact:
+    _installed_openstacksdk_versions: "{{ _installed_openstacksdk['packages']['pip']['openstacksdk'] | map(attribute='version') | list }}"
 
 - name: print the current installed version of openstacksdk
   ansible.builtin.debug:
-    var: _installed_openstacksdk.stdout
+    var: _installed_openstacksdk_versions[0]
```

Comment:
``_installed_openstacksdk_versions | first`` might be considered slightly more elegant... but meh